src/vivarium/framework/results/manager.py

```diff
@@ -126,19 +102,21 @@ def on_post_setup(self, _: Event) -> None:
                 )
                 for observation in observations:
                     measure = observation.name
-                    df = observation.results_initializer(
+                    self._raw_results[measure] = observation.results_initializer(
                         event_requested_stratification_names, registered_stratifications
                     )
-                    self._raw_results[measure] = df
                     if observation.stratifications is not None:
-                        unused_stratifications = self._track_stratifications(
+                        used_stratifications = self._track_stratifications(
                             measure,
                             event_requested_stratification_names,
                             registered_stratification_names,
                             missing_stratifications,
-                            unused_stratifications,
+                            used_stratifications,
                         )
 
+        unused_stratifications = registered_stratification_names.difference(
```

Comment:
This works but isn't something like this simpler?
```python
used_stratifications = set()
for event_name in self._results_context.observations:
    for _, stratifications in self._results_context.observations[event_name]:
        used_stratifications |= stratifications

unused_stratifications = registered_stratifications - used stratifications
missing_stratifications = used_stratifications - registered_stratifications
```

We lose information about the specific observers that are requesting unregistered stratifications, but I'm not sure how important that information is. It should be pretty easy for a user to track down in any case.

Reply:
Isn't that exactly what I'm doing (only it's happening in the `_track_stratifications` helper function)?

Reply:
Wait, I think I misread what you were saying

Reply:
Ok, yeah I see what you're saying. Yes, if you're ok w/ losing the measure-specific info then I agree this is much better. Done